ui/StatusQ/include/StatusQ/sumaggregator.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <QAbstractListModel>
+#include <QObject>
+#include <QVariant>
+#include <QDebug>
+
+#include "StatusQ/aggregator.h"
+
+class SumAggregator : public Aggregator {
+
+public:
+    explicit SumAggregator(QObject *parent = nullptr) {}
+
+private slots:
+    virtual void calculateAggregation() override {
+        m_value = 0;
+
+        // Check if m_model exists
+        if (!m_model) {
+            qWarning() << "Model does not exist";
+            emit valueChanged();
+            return;
+        }
+
+        // Check if m_roleName is part of the roles of the model
+        QHash<int, QByteArray> roles = m_model->roleNames();
+        if (!roleExists()) {
+            qWarning() << "Provided role name does not exist in the current model";
+            emit valueChanged();
+            return;
+        }
+
+        float total = 0.0f;
+        int rows = m_model->rowCount();
+        int role = roles.key(m_roleName.toUtf8());
+
+        for (int row = 0; row < rows; ++row) {
+            QModelIndex index = m_model->index(row, 0);
+            QVariant value = m_model->data(index, role);
+
+            // Check if the value is a QString
```

Comment:
Why is this explicit check for the type being a string needed?

Reply:
I mean, it's a QVariant, you can just check whether it converts `toDouble()` (as you do below), that's all you need

Reply:
My intention here was to potentially expect both types, a double or a string. That's why this implementation.